### ISSUE TITLE
How to get metrics like "kubelet_volume_stats_used_bytes" and "kubelet_volume_stats_available_bytes" to get PVC usage in k3s cluster?

### ISSUE THREAD

**Issue by dsolanki28** [k3s-io/k3s#6962]
We would like to have metrics "kubelet_volume_stats_used_bytes" and "kubelet_volume_stats_available_bytes" available in prometheus to monitor PVC usage in k3s cluster. To enable them, we tried to install k3s cluster with arguements "--kubelet-arg kubelet-preferred-address-types=InternalIP,Hostname,ExternalIP --kubelet-arg enable-kubelet-volume-stats" but error showing the flags are unknown to k3s.

Request you please help us to enable these metrics, or other supported metrics to get PVC usage in k3s cluster.

**Comment by brandond** [comment]
This is an issue with Kubernetes/cadvisor when used with containerd (or pretty much any other CRI backend now that dockershim has been removed). You can track it upstream at https://github.com/google/cadvisor/issues/2785#issuecomment-1205538108

**Comment by micke** [comment]
@brandond Is this still the case? I thought `kubelet_volume_stats_used_bytes` was gathered by kubelet without involving cadvisor? I've tried to understand what metrics come from where but i'm finding it a bit hard to get a understanding of it all.

I'm using K3S 1.34.1+k3s1 with containerd 2.14 installed throught the OS, not containerd embedded in k3s.

I'm mostly using PVC's through openebs-zfs, but to my understanding it should support the metrics: https://github.com/openebs/zfs-localpv/blob/develop/pkg/driver/agent.go#L406:L445

I've also created a PVC with using `storageClassName: local-path` but i'm not seeing any stats for that either:

```bash
$ kubectl get --server https://<nodename>:10250 --raw /metrics | grep kubelet_volume
# HELP kubelet_volume_metric_collection_duration_seconds [ALPHA] Duration in seconds to calculate volume stats
# TYPE kubelet_volume_metric_collection_duration_seconds histogram
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="du",le="0.005"} 1.339005e+06
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="du",le="0.01"} 1.339005e+06
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="du",le="0.025"} 1.339936e+06
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="du",le="0.05"} 1.339969e+06
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="du",le="0.1"} 1.339969e+06
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="du",le="0.25"} 1.339969e+06
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="du",le="0.5"} 1.339969e+06
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="du",le="1"} 1.339969e+06
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="du",le="2.5"} 1.339969e+06
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="du",le="5"} 1.339969e+06
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="du",le="10"} 1.339969e+06
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="du",le="+Inf"} 1.339969e+06
kubelet_volume_metric_collection_duration_seconds_sum{metric_source="du"} 32.72115083700014
kubelet_volume_metric_collection_duration_seconds_count{metric_source="du"} 1.339969e+06
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="statfs",le="0.005"} 13
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="statfs",le="0.01"} 13
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="statfs",le="0.025"} 13
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="statfs",le="0.05"} 13
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="statfs",le="0.1"} 13
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="statfs",le="0.25"} 13
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="statfs",le="0.5"} 13
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="statfs",le="1"} 13
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="statfs",le="2.5"} 13
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="statfs",le="5"} 13
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="statfs",le="10"} 13
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="statfs",le="+Inf"} 13
kubelet_volume_metric_collection_duration_seconds_sum{metric_source="statfs"} 0.000201024
kubelet_volume_metric_collection_duration_seconds_count{metric_source="statfs"} 13
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="zfs.csi.openebs.io",le="0.005"} 18602
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="zfs.csi.openebs.io",le="0.01"} 22111
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="zfs.csi.openebs.io",le="0.025"} 22118
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="zfs.csi.openebs.io",le="0.05"} 22118
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="zfs.csi.openebs.io",le="0.1"} 22118
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="zfs.csi.openebs.io",le="0.25"} 22118
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="zfs.csi.openebs.io",le="0.5"} 22118
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="zfs.csi.openebs.io",le="1"} 22118
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="zfs.csi.openebs.io",le="2.5"} 22118
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="zfs.csi.openebs.io",le="5"} 22118
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="zfs.csi.openebs.io",le="10"} 22118
kubelet_volume_metric_collection_duration_seconds_bucket{metric_source="zfs.csi.openebs.io",le="+Inf"} 22118
kubelet_volume_metric_collection_duration_seconds_sum{metric_source="zfs.csi.openebs.io"} 92.65292817100013
kubelet_volume_metric_collection_duration_seconds_count{metric_source="zfs.csi.openebs.io"} 22118
```

**Comment by micke** [comment]
In my case it looks like it was a regression introduced with kubernetes 1.34.1 that is fixed in 1.34.2: https://github.com/kubernetes/kubernetes/issues/133847
